### ISSUE TITLE
Theia Editor Key Bindings switching (Default, VI, Emacs)

### ISSUE THREAD

**Issue by vparfonov** [eclipse-che/che#9244]
Provide ability for changing editor Key Bindings (Default, VI, Emacs)

**Comment by RomanNikitenko** [comment]
the plugin with Emacs key bindings is temporarily placed [here](https://github.com/RomanNikitenko/che-theia-emacs-plugin/pull/1)

**Comment by RomanNikitenko** [comment]
Corresponding to [this comment](https://github.com/theia-ide/theia/blob/da4e14af3a1b8ac24d7ec3368161d70b225c8dd4/packages/monaco/src/browser/monaco-keybinding.ts#L44) Theia does not support chord keybindings properly at the moment, so keybindings for actions like 'Delete blank lines around'(Ctrl+X Ctrl+O) or 'Select All'(Ctrl+X H) are not implemented.

**Comment by RomanNikitenko** [comment]
depends on https://github.com/theia-ide/theia/issues/2783

**Comment by epnasis** [comment]
Can you share update on the plans for introducing VI bindings?

**Comment by RomanNikitenko** [comment]
Our first attempt to provide VI and Emacs key bindings was providing own extensions.  
Some set of key bindings was implemented within this work.

At some stage of our project development we decided to change approach and invest our efforts for vs code extensions support - it allows to use existed extensions/plugins.
So the work on the issue was stopped till plugin API is aligned with vs code, vs code format of context is supported, sections from `package.json` of an extensions is used, commands Id is aligned with vs code commands and so on.

I tried to turn on VI and Emacs vs code extensions for theia a few days ago. Looks like theia picks up the key bindings from the extensions

![vim](https://user-images.githubusercontent.com/5676062/62936223-ec319e00-bdd1-11e9-9b48-7942707d7308.png)

But there are still a lot of conflicts:
![errors_keybindings](https://user-images.githubusercontent.com/5676062/62936235-f8b5f680-bdd1-11e9-9105-922689256a41.png)

and I guess we have other incompatibilities.

Unfortunately it still works incorrectly, so something is missing, we should investigate it and define what we need to align with vs code to support these extensions.
